File: OpsService/schemas/model_config.py
from pydantic import BaseModel
from datetime import datetime

class ModelConfigCreate(BaseModel):
    name: str
    provider: str
    model_name: str
    api_key: str
    base_url: str | None = None
    is_default: bool = False
    temperature: float = 0.7
    embedding_model: str | None = None
    embedding_dimensions: int | None = None
  
class ModelConfigUpdate(BaseModel):
    name: str | None = None
    provider: str | None = None
    model_name: str | None = None
    api_key: str | None = None
    base_url: str | None = None
    is_default: bool | None = None
    temperature: float | None = None
    embedding_model: str | None = None
    embedding_dimensions: int | None = None

class ModelConfigOut(BaseModel):
    id: int
    name: str
    provider: str
    model_name: str
    embedding_model: str | None
    embedding_dimensions: int | None
    base_url: str | None
    is_default: bool
    temperature: float | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: Frontend/src/api/modelConfig.ts
import { get, post, put, del } from './OpsRequest'

// --- 类型定义 ---

export interface ModelConfig {
  id: number
  name: string
  provider: string
  model_name: string
  base_url: string | null
  is_default: boolean
  temperature: number
  created_at: string
  updated_at: string
}

export interface ModelConfigCreateData {
  name: string
  provider: string
  model_name: string
  api_key: string
  base_url?: string
  is_default?: boolean
  temperature?: number
}

export type ModelConfigUpdateData = Partial<Omit<ModelConfigCreateData, 'api_key'>> & {
  api_key?: string
}

// --- ModelConfig API ---

export const getModelConfigs = () =>
  get<ModelConfig[]>('/api/v1/model-configs')

export const getModelConfig = (id: number) =>
  get<ModelConfig>(`/api/v1/model-configs/${id}`)

export const createModelConfig = (data: ModelConfigCreateData) =>
  post<ModelConfig>('/api/v1/model-configs', data)

export const updateModelConfig = (id: number, data: ModelConfigUpdateData) =>
  put<ModelConfig>(`/api/v1/model-configs/${id}`, data)

export const deleteModelConfig = (id: number) =>
  del(`/api/v1/model-configs/${id}`)
